src/commands.cpp

```diff
@@ -1959,50 +1959,87 @@ void CommandUserRequest::procresult()
     client->app->invite_result(e);
 }
 
-CommandPutUA::CommandPutUA(MegaClient* client, const char *an, const byte* av, unsigned avl)
+CommandPutUA::CommandPutUA(MegaClient* client, const char *an, const byte* av, unsigned avl, int ctag)
 {
-    cmd("up");
+    this->an = an;
+    this->av.assign((const char*)av, avl);
+    this->u = client->ownuser();
 
-    // if removing avatar, do not Base64 encode the attribute value
-    if (!strcmp(an, "+a") && !strcmp((const char *)av, "none"))
-    {
-        arg(an,(const char *)av, avl);
-    }
-    else
+    cmd("upv");
+
+    beginarray(an);
     {
-        arg(an, av, avl);
+        // if removing avatar, do not Base64 encode the attribute value
+        if (!strcmp(an, "+a") && !strcmp((const char *)av, "none"))
+        {
+            element((const char*)av);
+        }
+        else
+        {
+            element(av, avl);
+        }
+
+        const string *attrv = u->getattrversion(an);
+        if (attrv)
+        {
+            element(attrv->c_str());
+        }
     }
+    endarray();
 
-    tag = client->reqtag;
+    tag = ctag;
 }
 
 void CommandPutUA::procresult()
 {
-    error e;
-
     if (client->json.isnumeric())
     {
-        e = (error)client->json.getint();
+        client->app->putua_result((error)client->json.getint());
     }
     else
     {
-        client->json.storeobject();
-        e = API_OK;
-    }
+        const char* ptr;
+        const char* end;
 
-    client->app->putua_result(e);
+        if (!(ptr = client->json.getvalue()) || !(end = strchr(ptr, '"')))
+        {
+            client->app->putua_result(API_EINTERNAL);
+            return;
+        }
+        string an = string(ptr, (end-ptr));
+
+        if (!(ptr = client->json.getvalue()) || !(end = strchr(ptr, '"')))
+        {
+            client->app->putua_result(API_EINTERNAL);
+            return;
+        }
+        string v = string(ptr, (end-ptr));
+
+        if (an.empty() || v.empty() || (an != an))
```

Comment:
In the same way than previous comment, the response of `upv` can return more than one pair attribute_name-attribute_version. Since the SDK doesn't send multiple attributes at once, this `procresult()` only process one attribute.